app/lib/actions/katello/capsule_content/sync.rb

```diff
@@ -27,17 +27,31 @@ def plan(capsule_content, options = {})
               need_updates.each do |repo|
                 plan_action(Pulp::Repository::Refresh, repo, capsule_id: capsule_content.capsule.id)
               end
-
               plan_action(ConfigureCapsule, capsule_content, environment, content_view)
+              sync_repos_to_capsule(capsule_content, repository_ids)
+              plan_action(RemoveOrphans, :capsule_id => capsule_content.capsule.id)
+            end
+          end
+        end
 
-              concurrence do
-                repository_ids.each do |repo_id|
-                  plan_action(Pulp::Consumer::SyncCapsule,
+        def sync_repos_to_capsule(capsule_content, repository_ids)
+          concurrence do
+            repository_ids.each do |repo_id|
+              sequence do
+                repo = ::Katello::Repository.where(:pulp_id => repo_id).first ||
+                       ::Katello::ContentViewPuppetEnvironment.where(:pulp_id => repo_id).first
+                if repo && repo.content_type != "yum"
```

Comment:
I'm actually not sure it is okay to assume only puppet.  With yum we're relying on the 'remove_missing' feature of pulp that only the yum distributor supports.  Puppet is likely the only one that that will error without this, but i think the other types need it too